lib/puppet/provider/elastic_plugin.rb

```diff
@@ -155,6 +155,10 @@ def es_version
     es_save = ENV['ES_INCLUDE']
     java_save = ENV['JAVA_HOME']
 
+    if @resource[:java_home]
```

Comment:
You can probably drop this whole conditional as an `else` to the `os == 'OpenBSD'` check since that block of code is all about handling `JAVA_HOME`.